tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -120,17 +121,19 @@ def test_5_snapshots(
         microvm = microvm_factory.build()
         microvm.spawn()
         microvm.restore_from_snapshot(snapshot, resume=True)
+        vm.ssh.run("pkill -SIGCONT socat")
         # Test vsock guest-initiated connections.
         path = os.path.join(
             microvm.path, make_host_port_path(VSOCK_UDS_PATH, ECHO_SERVER_PORT)
         )
         check_guest_connections(microvm, path, vm_blob_path, blob_hash)
         # Test vsock host-initiated connections.
-        path = start_guest_echo_server(microvm)
+        path = os.path.join(microvm.jailer.chroot_path(), VSOCK_UDS_PATH)
         check_host_connections(path, blob_path, blob_hash)
 
         # Check that the root device is not corrupted.
         check_filesystem(microvm.ssh, "squashfs", "/dev/vda")
+        vm.ssh.run("pkill -SIGSTOP socat")
```

Comment:
Does `socat` do something special regarding `STOP`/`CONT`? AFAICS `socat` would be waiting on an `accept()` system call, so how are STOP/CONT making a difference?

Reply:
the socket connection is reset by Firecracker while creating a snapshot so it seems to leave socat in a weird state and without stop/cont 'ng socat before/after snapshot we end up in a vsock timeout when host tries to connect to guest. The PR unblocks the pipeline but we still need to dig deeper separately.

Reply:
I can confirm what sudanl0@ says.

Here is a run with the patch of Sudan but without SIGSTOP/SIGCONT pair: https://buildkite.com/firecracker/bchalios-ad-hoc-tests/builds/38#_
And here is with original patch: https://buildkite.com/firecracker/bchalios-ad-hoc-tests/builds/37

It looks as if vsock gets completely broken if we snapshot while something it's using it. I agree with Sudan we should merge this to fix the test and look into the vsock issue(?) separately.

Reply:
So is the issue lingering `socat` processes from previous runs? What happens if we kill those processes after the checks?

Reply:
I mean, it is a great find that the STOP/CONT makes it work. I am fine approving the PR, but we should understand why this works.

Could we add a comment saying why we do STOP/CONT, and that we don't know why it works? Maybe a `TODO`?

Reply:
I agree with adding a TODO and opening an issue for investigating

Reply:
Added a TODO but I would like to use existing ticket to debug the issue further instead of creating a new issue.